4_naming_conventions/43_file_names.md

```diff
@@ -0,0 +1,211 @@
+```
+<Feature><Phase>.inf
+
+   <Feature>      REQUIRED    *
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm, StandaloneMm,
+                              Smm, Uefi.
+
+Example:
+   - SmbiosDxe.inf
+```
+
+* If the implementation of CPU arch or vendor shares some code in the module. Or the implementation is only for the specific CPU arch or vendor:
+```
+<Feature><Phase><CpuArch><Vendor>.inf
+
+   <Feature>      REQUIRED    *
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm,
+                              StandaloneMm, Smm, Uefi.
+   <CpuArch>      OPTIONAL    The <CpuArch> is represented with a BNF,
+                              <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                        'LoongArch64' | 'Ebc'
+                              <CpuArch> ::= <arch>['_'<arch>]*
+                              
+                              Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>       OPTIONAL    *
+
+Example:
+   - CpuIo2DxeAmd.inf
+   - CpuIo2DxeIntel.inf
+   - CpuIo2DxeIa32_X64.inf
+
+```
+
+* If the implementation does not have any shared code between phases:
+
+```
+<Feature>/<Phase>/<Feature>.inf
+  or
+<Feature>/<Phase>/<Feature><CpuArch><Vendor>.inf
+
+   <Feature>      OPTIONAL    *
+   <Phase>        REQUIRED    Base, Sec, Pei, Dxe, DxeRuntime, Mm, StandaloneMm, Smm,
+                              Uefi.
+   <CpuArch>      OPTIONAL    The <CpuArch> is represented with a BNF,
+                              <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                        'LoongArch64' | 'Ebc'   
+                              <CpuArch> ::= <arch>['_'<arch>]*
+                              
+                              Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>       OPTIONAL    *
+Example:
+   PCD/Dxe/Pcd.inf
+```
+
+#### 4.4.5.3 EDK II INF file within a Library instance
+```
+<Phase><CpuArch><Vendor><LibraryClassName><Dependency>.inf
+   <Phase>              REQUIRED     Base, Sec, Pei, Dxe, DxeRuntime, Mm,
+                                     StandaloneMm, Smm, Uefi.
+   <CpuArch>            OPTIONAL     The <CpuArch> is represented with a BNF,
+                                     <arch> ::='Ia32' | 'X64' | 'Arm' | 'AArch64' | 'RiscV64' |
+                                               'LoongArch64' | 'Ebc'
+                                     <CpuArch> ::= <arch>['_'<arch>]*
+                              
+                                     Example: Ia32_X64_Arm or RiscV64_LoongArch64
+   <Vendor>             OPTIONAL     *
+   <LibraryClassName>   REQUIRED     *
+   <Dependency>         OPTIONAL     * (Typically name of PPI, Protocol, LibraryClass
+                                       that the implementation is layered on top of)
+
+Example:
+   - SmmAmdSmmCpuFeaturesLib.inf
+   - SmmIa32_X64SmmCpuFeaturesLib.inf
+```
+
+#### 4.3.5.4 EDK II source files within a Library/Module instance

```

Comment:
__FileName__ gives the freedom to developers when there is no \<CpuArch\> and \<Vendor\>, however, we should define the rules when considering the archs and vendors in the source file name. But I can remove the second part because it mixes up the file and directory naming.